pkg/kubelet/cm/deviceplugin/manager_test.go

```diff
@@ -85,13 +81,6 @@ func TestDevicePluginReRegistration(t *testing.T) {
 	// Wait for the first callback to be issued.
 
 	<-callbackChan
-	// Wait till the endpoint is added to the manager.
```

Comment:
this kind of sleep operation is a clear indication that the current lock implementation is buggy ...

Reply:
As i explained in the PR description there was a race because of the relative ordering between adding endpoint in the manager and invocation of first callback, which is not there anymore. Hence removing this sleep. I think it is unfair to blame current lock implementation for the explained race and that too when it does not exist anymore.

Reply:
+1 to what @vikaschoudhary16 has explained.